http3/server.go

```diff
@@ -265,6 +294,23 @@ func (s *Server) handleRequest(sess quic.Session, str quic.Stream, decoder *qpac
 		s.logger.Infof("%s %s%s", req.Method, req.Host, req.RequestURI)
 	}
 
+	// Adjust the read deadline if necessary
+	if setReadDeadline && !hdrDeadline.Equal(wholeReadDeadline) {
+		err := str.SetReadDeadline(wholeReadDeadline)
+		if err != nil {
+			return newStreamError(errorSettingsError, err)
+		}
+		s.logger.Debugf("Adjusted read deadline to the ReadTimeout on StreamID %v", str.StreamID())
+	}
+	// Done with header, now set the write timeout
+	if d := s.WriteTimeout; d != 0 {
+		err = str.SetWriteDeadline(time.Now().Add(d))
+		if err != nil {
+			return newStreamError(errorSettingsError, err)
```

Comment:
This should be an internal error.

Reply:
Of course.

Reply:
I saw that it's been used like [`_ = str.SetWriteDeadline(...)`](https://github.com/lucas-clemente/quic-go/blob/4ae1a1350368c2e21f4213b71faf492f487e14bf/stream.go#L139) as it would never return err  but I thought maybe that would be the future improvement to add error. Nevertheless, I changed it to internal error.

I can remove the error handling though ...